### PR TITLE
addinvoice: simplify usage of PopulateHopHints

### PR DESCRIPTION
**Pull request by sputn1ck**
This PR simplifies the usage of `PopulateHopHints` by removing the use of the `channeldb.OpenChannel` struct in the `ListChannels` func to the previously used `HopHintInfo`.

### CODE REVIEW

File: lnrpc/invoicesrpc/addinvoice.go [hunk 1]
```diff
@@ -629,11 +614,7 @@ type SelectHopHintsCfg struct {
 
 	// FetchAllChannels retrieves all open channels currently stored
```
Comment (sputn1ck) [inline lnrpc/invoicesrpc/addinvoice.go:615]: godoc would need to be updated

File: lnrpc/invoicesrpc/addinvoice.go [hunk 2]
```diff
@@ -831,3 +813,47 @@ func PopulateHopHints(cfg *SelectHopHintsCfg, amtMSat lnwire.MilliSatoshi,
...
+		)
+		hopHint := newHopHintInfoFromDbChannel(dbChannel, channelActive)
+		hopHints = append(hopHints, hopHint)
+	}
+
+	return hopHints, nil
+}
+
+// newHopHintInfoFromDbChannel creates a new HopHintInfo from the provided
+// channeldb.OpenChannel
```
Comment (sputn1ck) [inline lnrpc/invoicesrpc/addinvoice.go:843]: missing dot

File: lnrpc/invoicesrpc/addinvoice.go [hunk 3]
```diff
@@ -691,18 +676,17 @@ func getPotentialHints(cfg *SelectHopHintsCfg) ([]*channeldb.OpenChannel,
 		return nil, err
 	}
 
-	privateChannels := make([]*channeldb.OpenChannel, 0, len(openChannels))
+	privateChannels := make([]*HopHintInfo, 0, len(openChannels))
 	for _, oc := range openChannels {
-		isPublic := oc.ChannelFlags&lnwire.FFAnnounceChannel != 0
-		if !isPublic {
+		if !oc.IsPublic {
```
Comment (positiveblue) [inline lnrpc/invoicesrpc/addinvoice.go:681]: nit: now that we already have the `HopHintInfo` here maybe we can do `if !oc.IsPublic && oc.IsActive {`

File: lnrpc/invoicesrpc/addinvoice.go [hunk 4]
```diff
@@ -713,27 +697,25 @@ func getPotentialHints(cfg *SelectHopHintsCfg) ([]*channeldb.OpenChannel,
 // shouldIncludeChannel returns true if the channel passes all the checks to
 // be a hopHint in a given invoice.
 func shouldIncludeChannel(cfg *SelectHopHintsCfg,
```
Comment (positiveblue) [inline lnrpc/invoicesrpc/addinvoice.go:699]: nit: rename to `shouldIncludeHopHint`?

File: lnrpc/invoicesrpc/addinvoice.go [hunk 5]
```diff
@@ -713,27 +697,25 @@ func getPotentialHints(cfg *SelectHopHintsCfg) ([]*channeldb.OpenChannel,
...
-	channel *channeldb.OpenChannel,
+	channel *HopHintInfo,
```
Comment (positiveblue) [inline lnrpc/invoicesrpc/addinvoice.go:700]: nit: `hopHintInfo`?

**Review by positiveblue — CHANGES_REQUESTED** [review]
The changes LGTM + will make it easier for other applications to use this method with `lndclient`

**Pull request closed without merging**